EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js

```diff
@@ -1,19 +1,55 @@
 // Main class for all android/ios app modifications (assuming cordova)
+var endTime, startTime;
+function calculateTime() {
+
+    var url = window.location.pathname.split("/");
+    url.pop(); // index.html
+    var book = url.pop();
+    var subject = url.pop();
+    var moduleName = url.pop();
+
+    // remove the special characters found in the URL string
+    function removeSpecialChars(str) {
+        return str.replace(/%20/g, "    ");
+    }
+    var moduleName = removeSpecialChars(moduleName);
+    var levelName = removeSpecialChars(subject);
+    var lessonName = removeSpecialChars(book);
+
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds
+    timeDiff /= 1000;
+    var elapsedTime = Math.round(timeDiff);
+
+    document.addEventListener("deviceready", function () {
+        window.FirebasePlugin.setAnalyticsCollectionEnabled(true);
+        window.FirebasePlugin.setScreenName("Module_Play_Status");
+        // Log only the module name for the Epic Quest section
+        if (lessonName == "Epic    Quest") {

```

Comment:
Is the lesson name "Epic    Quest" along with these spaces?

Reply:
Removed the spaces.